### PR TITLE
feat(scheduled-tasks): add MCP pre-approval UI

### PR DESCRIPTION
**Pull request by wenxi-onyx**
## Description

Scheduled task pre-approval UI

Add MCP servers to the scheduled-task create and edit picker. Show both target types in the task detail summary.

App and MCP selections remain independent. Unavailable MCP grants remain visible and removable.

The picker sections fill the container width. They use a responsive two-column layout when space permits.

This PR is stacked on #13884, which adds the backend API and storage support.

## How Has This Been Tested?

- Tested the happy path and edge cases end to end with the local kind cluster.
- Verified create, detail, edit, removal, invalid IDs, and cleanup.
- Verified full-width desktop and mobile layouts with Playwright.
- Passed 10 focused Jest tests across three suites.
- Passed format, lint, type check, and secret scan commit hooks.

## Additional Options

- [ ] [Optional] Please cherry-pick this PR to the latest release version.
- [x] [Optional] Override Linear Check

### CODE REVIEW

File: backend/tests/external_dependency_unit/craft/test_scheduled_task_pre_approvals.py [hunk 1]
```diff
@@ -23,13 +23,12 @@
     ScheduledTaskStatus,
```
[the remark this answers is not in the document]
Comment (wenxi-onyx) [reply backend/tests/external_dependency_unit/craft/test_scheduled_task_pre_approvals.py:23]: Addressed in `9696194`. We replaced the private-helper test with API integration coverage. `test_create_and_patch_mcp_pre_approvals` exercises deduplication and unavailable-ID rejection through HTTP. `test_patch_retains_and_removes_existing_unavailable_mcp_pre_approval` covers the existing-grant carve-out and removal. This keeps the behavior covered at the public boundary.

**Review by Danelegend — APPROVED** [review]

**Comment by wenxi-onyx** [comment]
@greptile @codex re-review
